src/dataset.jl

```diff
@@ -5,49 +5,23 @@
 const RECORDINGS_FILE_NAME = "recordings.msgpack.zst"
 
 struct Dataset
-    path::String
+    path::Any
```

Comment:
Why not make this a type parameter to ensure `Dataset` is fully concrete?

Reply:
To prevent downstream code from using it for dispatch (also any potential dynamic dispatch hit here is super negligible compared to e.g. hitting disk/network requests). I actually had a comment to explain this in an earlier draft, I should probably add that back so it's obvious

Reply:
> To prevent downstream code from using it for dispatch

Why would that be a bad thing?

Reply:
If downstream code wants to specialize on the path, they should do so at the Path API layer. If it were a type parameter, somebody might get confused and try to do so at the Dataset API layer instead, which should generally remain compositionally orthogonal to storage layer stuff